### PR TITLE
Refactor performance tests: capture memory usage, add README

### PR DESCRIPTION
**Pull request by jdn5126**
**What type of PR is this?**
enhancement, testing

**Which issue does this PR fix**:
https://github.com/aws/amazon-vpc-cni-k8s/issues/2437

**What does this PR do / Why do we need it**:
This PR refactors and cleans up `scripts/run-integration-tests.sh`, primarily for performance testing. It also captures memory usage of the `aws-node` pod during performance tests and uploads it to an S3 bucket. It also adds a README for running `run-integration-tests.sh` and cleans up unused variables.

The usage of the memory statistics may be extended in the future, i.e. to compare to previous runs, but for now we only flag when memory usage exceeds our recommended value: 200Mi.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:
Manually running integration tests and evaluating performance statistics.

**Automation added to e2e**:
Added capturing of memory usage by `aws-node` pods to performance tests.

**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No, Yes

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
No

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/weekly-cron-tests.yaml [hunk 1]
```diff
@@ -39,55 +39,47 @@ jobs:
...
           RUN_BOTTLEROCKET_TEST: true
           RUN_TESTER_LB_ADDONS: true
-          RUN_INTEGRATION_DEFAULT_CNI: false
         run: |
           ./scripts/run-integration-tests.sh
         if: always()
       - name: Run calico tests
         env:
           DISABLE_PROMPT: true
-          S3_BUCKET_CREATE: false
```
Comment (orsenthil) [inline .github/workflows/weekly-cron-tests.yaml:83]: What's the motivation of these changes, especially the removal of S3 bucket creates and setting RUN_INTEGRATION_DEFAULT_CNI to false in the weekly cron?

**Review by orsenthil — APPROVED** [review]
LGTM. Could you please paste an output of the test run after this refactor. It will be good to verify what what to expect, and see a successful run. 


Thank you.

File: .github/workflows/weekly-cron-tests.yaml (hunk 1 above)
Comment (jdn5126) [reply .github/workflows/weekly-cron-tests.yaml:83]: `S3_BUCKET_CREATE` and `S3_BUCKET_NAME` were unused variables that appear to have been missed in a previous cleanup, so I removed them here.

For `RUN_INTEGRATION_DEFAULT_CNI`, we were running the CNI integration tests with every invocation of `run-integration-tests`. For the weekly tests, that means we were running them 4 times with no real benefit. Since they already run nightly as part of the `nightly-cron-tests`, I concluded they should be removed from the weekly tests. This should drop our weekly test runtime by about 1.5 hours, as well.

**Comment by jdn5126** [comment]
For test results, I kicked off https://github.com/aws/amazon-vpc-cni-k8s/actions/runs/5488008758 to show the results from GitHub

**Comment by jdn5126** [comment]
> For test results, I kicked off https://github.com/aws/amazon-vpc-cni-k8s/actions/runs/5488008758 to show the results from GitHub

Oh, actually that won't work, as the run cannot happen with the new workflow file until this merges. For example output, I can show the latest S3 bucket results from my manual run:
```
Iteration, scaleUpTime (s), scaleDownTime (s), scaleUpMem (Mi), scaleDownMem (Mi)
1, 13, 25, 57, 59
2, 13, 26, 59, 59
3, 13, 25, 58, 59
```
